Phase2

arsany98 opened the pull request.

arsany98 committed:
feat: Create Tabs, DbContext, Routing and Skeleton
- Create DbContext using json server.
- Change the courses_db.json file.
- Create Routes for course page.
- Create Skeleton for next components.

arsany98 committed:
feat: Create Home Page, Navbar and Footer.
- Create Navbar nad header elements including search form

arsany98 committed:
feat: Create WithData HOC, Course Page Components

arsany98 committed:
feat: Create Course Page.

arsany98 committed:
fix: Use Id in key and other fixes
- Use bigger instructors images
- Create show more for extra caption languages

arsany98 committed:
fix: course content length, instructor image in db

Elwazer007 reviewed (changes requested):
when selecting a couse that has no data included i 'm getting a blank , page 
either include its data or add some error handling

Elwazer007 reviewed (commented):
refactor the navbar component , it's fat exceeding 100 lines of code for example the search bar shloud be in a seperate component here

arsany98 committed:
fix: Use self enclosing tags, use color variables
- Use self enclosing tags if the component doesn't have children
- Use color variables instead of hardcoded color in some places

arsany98 committed:
fix: Add skeleton, useMemo and other changes
- When a course has no data the course page shows skeleton instead
- Filter courses in useMemo hook
- Create Not found Page
- Change fetch error message to be more descriptive

arsany98 committed:
fix: Split some components and other fixes
- Create SearchForm, FixedBar, Incentives components
- Correct Incentive spelling error
- Remove App.css

arsany98 committed:
fix: Fix fetching twice and other fixes
- Use loaded boolean to prevent fetching twice
- Add padding bottom to footer
- Adjust height of show more button in CourseHeader

arsany98 committed:
refactor: Change project structure and other.
- Split videoInfoList to seperate component
- Add dummy Data for courses

arsany98 committed:
refactor: Create getInstructorsNames util
- Use getInstructorsNames in CourseHeader

arsany98 committed:
fix: Fix course_data id
